samples/python/onedim/diffusion_flame_continuation.py

```diff
@@ -54,43 +62,139 @@
+
+for i in range(Nmax):
+    if strain_rate > 0.98 * a_max:
+        spacing = initial_spacing
+    else:
+        spacing = unstable_spacing
     control_temperature = np.min(f.T) + spacing*(np.max(f.T) - np.min(f.T))
-    print(f'Iteration {i}: Control temperature = {control_temperature} K')
+
+    # Store the flame state in case the iteration fails and we need to roll back
+    backup_state = f.to_array()
+
+    logger.debug(f'Iteration {i}: Control temperature = {control_temperature:.2f} K')
     f.set_left_control_point(control_temperature)
     f.set_right_control_point(control_temperature)
 
     # This decrement is what drives the two-point control. If failure
     # occurs, try decreasing the decrement.
     f.left_control_point_temperature -= temperature_increment
     f.right_control_point_temperature -= temperature_increment
+    f.clear_stats()
+
+    if (f.left_control_point_temperature < f.fuel_inlet.T + 100
+        or f.right_control_point_temperature < f.oxidizer_inlet.T + 100
+    ):
+        logger.info("Stopping because control point temperature is sufficiently close "
+                    "to inlet temperature.")
+        break
 
     try:
         f.solve(loglevel=0)
-    except ct.CanteraError as e:
-        print('Solver did not converge. Stopping.')
+        if abs(max(f.T) - Tmax) < 0.8 * target_delta_Tmax:
+            # Max temperature is changing slowly. Try a larger increment next step
+            temperature_increment = min(temperature_increment + 3, max_increment)
+        elif abs(max(f.T) - Tmax) > target_delta_Tmax:
+            # Max temperature is changing quickly. Scale down increment for next step
+            temperature_increment *= 0.9 * target_delta_Tmax / (abs(max(f.T) - Tmax))
+        error_count = 0
+    except ct.CanteraError as err:
+        logger.debug(err)
+        if strain_rate / a_max < strain_rate_tol:
+            logger.info('Traversed unstable branch down to '
+                f'{100 * strain_rate / a_max:.1f}% of the maximum strain rate.')
+            break
+
+        # Restore the previous solution and try a smaller temperature increment for the
+        # next iteration
+        f.from_array(backup_state)
+        temperature_increment = 0.7 * temperature_increment
+        error_count += 1
+        logger.warning(f"Solver did not converge on iteration {i}. Trying again with "
+                       f"dT = {temperature_increment:.2f}")
+
+    if ct.hdf_support():
+        f.save(output_path / 'flame_profiles.h5', name=f'iteration{i}', overwrite=True)
+
+    # Collect output stats
+    Tmax = max(f.T)
+    Tmid = 0.5 * (min(f.T) + max(f.T))
+    s = np.where(f.T > Tmid)[0][[0,-1]]
+    width = f.grid[s[1]] - f.grid[s[0]]
+    strain_rate = f.strain_rate('max')
+    a_max = max(strain_rate, a_max)
+
+    data.append({
+        'Tmax': max(f.T),
+        'strain_rate': strain_rate,
+        'heat_release_rate': np.trapz(f.heat_release_rate, f.grid),
```

Comment:
I ended up running the code in order to see what is visualized. This line raises a deprecation warning:
```
DeprecationWarning: `trapz` is deprecated. Use `trapezoid` instead, or one of the numerical integration functions in `scipy.integrate`.
  'heat_release_rate': np.trapz(f.heat_release_rate, f.grid),
```

Reply:
`np.trapezoid` is only available in NumPy 2.0 and newer, so we're just dealing with the deprecation warnings until we drop support for Numpy 1.x. This applies to several other places where `np.trapz` is used. In the CI and test suite, we explicitly silence this warning.